apps/condo/domains/resident/tasks/sendBillingReceiptsAddedNotificationForOrganizationContextTask.js

```diff
@@ -73,7 +74,7 @@ const prepareAndSendNotification = async (keystone, context, receipt, resident,
         meta: { dv: 1, data },
         sender: { dv: 1, fingerprint: 'send-billing-receipts-added-notifications' },
         uniqKey: notificationKey,
-        organization: { id: resident.organization },
+        organization: organizationId && { id: organizationId },
```

Comment:
What if no organizationId?
Maybe you meant this?
```suggestion
        organization: organizationId && { id: organizationId },
        ...(organizationId ? { id: organizationId } : {}),
```

Reply:
This returns undefined/null or { id: organizationId }, this is not required field and can be undefined/null  or { id: organizationId } but not { id: undefined/null }

Reply:
Its 3 separate behaviours:

1. If some filter is `undefined`, keystone will try to ignore it. So it basically the "allow all" filter
2. If some filter is `organization: null` this will throw error
3. And if id is passed, then it resolves correctly

Reply:
In code above you're relying on "Keystone magic that live under the hood", please specify correct behaviour explicitly

Reply:
This is sent to sendMessageService, here 'organization' property is not required, and it does 
```
if (organization) { attributes['organization'] = { connect: organization } }
```
So undefined will not allow all, null will not throw